Include/internal/pycore_code.h

```diff
@@ -477,13 +473,9 @@ write_location_entry_start(uint8_t *ptr, int code, int length)
 #define ADAPTIVE_COOLDOWN_VALUE 52
 #define ADAPTIVE_COOLDOWN_BACKOFF 0
 
-#define MAX_BACKOFF_VALUE (16 - ADAPTIVE_BACKOFF_BITS)
-
-
 static inline uint16_t
 adaptive_counter_bits(uint16_t value, uint16_t backoff) {
```

Comment:
Maybe drop the `adaptive_counter` functions and replace them with the `backoff_counter` equivalents?

To do that cleanly we will probably need to change `_Py_CODEUNIT` to:
```C
typedef union {
    uint16_t cache;
    struct {
        uint8_t code;
        uint8_t arg;
    } op;
    backoff_counter_t counter;
} _Py_CODEUNIT;
```

Reply:
I decided to keep them and not expose the backoff counter structure to other places.

Reply:
I could dispense with `adaptive_counter_bits()`, using `make_backoff_counter()` directly below, but I would oppose getting rid of the `cooldown()` and `warmup()` helpers, because they are used in several/many places.